C++/Encoder Feedback Device/src/main/cpp/Robot.cpp

```diff
@@ -7,60 +7,71 @@
 
 #include <frc/TimedRobot.h>
 #include <frc/smartdashboard/smartdashboard.h>
-#include "rev/CANSparkMax.h"
-
+#include "rev/SparkMax.h"
+#include "rev/config/SparkMaxConfig.h"
+#include "rev/config/EncoderConfig.h"
 
 
 class Robot : public frc::TimedRobot {
 
+  /**
+   * Change these parameters to match your setup
+   */
+  static constexpr int kDeviceID = 1;
+  static constexpr auto kMotorType = rev::spark::SparkMax::MotorType::kBrushless;
 
-  // initialize SPARK MAX with CAN ID
-  static const int deviceID = 0;
-  rev::CANSparkMax m_motor{deviceID, rev::CANSparkMax::MotorType::kBrushed};
+  // Initialize the SPARK MAX with device ID and motor type
+  rev::spark::SparkMax m_motor{ kDeviceID, kMotorType };
 
   /**
-   * A CANEncoder object is constructed using the GetEncoder() method on an 
-   * existing CANSparkMax object. The assumed encoder type is the hall effect,
-   * or a sensor type and counts per revolution can be passed in to specify
-   * a different kind of sensor. Here, it's a quadrature encoder with 4096 CPR.
+   * A SparkRelativeEncoder object is constructed using the GetEncoder() method on an
+   * existing SparkMax object.
    */
-  rev::SparkRelativeEncoder m_encoder = m_motor.GetEncoder(rev::SparkRelativeEncoder::Type::kQuadrature, 4096);
+  rev::spark::SparkRelativeEncoder m_encoder = m_motor.GetEncoder();
 
   /**
-   * In order to use PID functionality for a controller, a CANPIDController object
-   * is constructed by calling the GetPIDController() method on an existing
-   * CANSparkMax object. 
-   * 
+   * In order to use PID functionality for a controller, a SparkClosedLoopController
+   * object is constructed by calling GetClosedLoopController() on an existing
+   * SparkMax object.
    */
-  rev::SparkPIDController m_pidController = m_motor.GetPIDController();
+  rev::spark::SparkClosedLoopController m_pidController = m_motor.GetClosedLoopController();
 
   // PID coefficients
   double kP = 0.1, kI = 1e-4, kD = 1, kIz = 0, kFF = 0, kMaxOutput = 1, kMinOutput = -1;
 
- public:
+public:
   void RobotInit() {
+    rev::spark::SparkMaxConfig maxConfig;
+
     /**
-     * The RestoreFactoryDefaults method can be used to reset the configuration parameters
-     * in the SPARK MAX to their factory default state. If no argument is passed, these
-     * parameters will not persist between power cycles
+     * The assumed encoder type is quadrature and, here, we set counts per
+     * revolution 4096.
      */
-    m_motor.RestoreFactoryDefaults();
+    maxConfig.encoder.CountsPerRevolution(4096);
 
     /**
-     * The PID Controller can be configured to use the analog sensor as its feedback
-     * device with the method SetFeedbackDevice() and passing the PID Controller
-     * the CANEncoder object. 
+     * By default, the PID controller will use the Hall sensor from a NEO for its
+     * feedback device. Instead, we can set the feedback device to the primary
+     * encoder object.
```

Comment:
It actually defaults to a feedback sensor of `none`. We 
```suggestion
     * Set the feedback device to the primary encoder object.
```